mantidproject/mantid
# Add project save to EngDiffUI

**Description of work.**
Add project save capability to the Engineering Diffraction UI, insofar as the data from the fitting tab is preserved.

**To test:**
1. In the fitting tab, load a focused run file or two (one on this PR https://github.com/mantidproject/mantid/pull/29976)
2. Plot it and run a fit with a few functions, peaks etc
3. File -> Save project
4. Close the interface, clear the ADS
5. Load the project from file
6. Check to see that the tab has been restored properly - runs in the table, workspaces plotted, fit function etc
7. Try the same but crash mantid using the segfault algorithm, and test that it reloads correctly (need to wait for project save to trigger first, this is printed in the logger when it happens)

Fixes #29830 . 





---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
